src/components/SignInButton.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import { useSearchParams } from "next/navigation";
+import { signIn } from "next-auth/react";
+import { ReactNode } from "react";
+
+type PropsType = {
+  children: ReactNode
+  provider: string
+}
+
+const SignInButton = ({children, provider}: PropsType) => {
+  const searchParams = useSearchParams();
+  const callbackUrl  = searchParams.get("callbackUrl");
```

Comment:
I think we should put this on the page rater than directly on the button.

Also, I think we don need to pass a callbackUrl on this `signIn` function